### ISSUE TITLE
The meaning of cls_loss2?

### ISSUE THREAD

**Issue by fyw1999** [Hzzone/PseCo#6]
Thank you for your excellent work and open source! I notice that another loss named  cls_loss2 is computed during training, which is not introduced in paper. I'm curious about the effect of cls_loss2, can you explain it? Thank you!

`with torch.autograd.set_grad_enabled(True) and torch.autocast(device_type='cuda', enabled=amp):
        cls_outs = cls_head(features, anchor_boxes, query_features)

        cls_loss = F.binary_cross_entropy_with_logits(cls_outs, query_labels, reduction='none')
        loss_mask = (query_labels >= 0).float()
        cls_loss = (cls_loss * loss_mask).sum() / (loss_mask.sum() + 1e-5)

        cls_outs2 = cls_head(features_a, clip_boxes, clip_target_features)
        cls_loss2 = F.binary_cross_entropy_with_logits(cls_outs2, clip_query_labels, reduction='none')
        loss_mask = (clip_query_labels >= 0).float()
        cls_loss2 = (cls_loss2 * loss_mask).sum() / (loss_mask.sum() + 1e-5)

        loss = cls_loss + cls_loss2 * cls_loss2_weight

        update_params = (n_iter % acc_grd_step == 0)
        loss = loss / acc_grd_step
        scaler(loss, optimizer=optimizer, update_grad=update_params)`

**Comment by Hzzone** [comment]
It is Eq. 3 in the paper.

**Comment by fyw1999** [comment]
> It is Eq. 3 in the paper.

Thanks for your reply! I understand your point, but I still have two questions.

(i) In the cls_loss (Eq. 2), the feature map of the bounding box region extracted by SAM's image encoder passes through two linear layers, then is multiplied with the feature map of the example bounding box extracted by CLIP's image encoder, and finally input to the BCE loss function to calculate the cls_loss. My question is, can better parameters of the ROIHead be learned by multiplying the feature map of the predicted bounding box with the fixed feature map of the example bounding box?

(ii) In the cls_loss2 (Eq. 3), for the five boxes of each predicted point, the SAM feature and CLIP feature of each box are extracted respectively, and the product is calculated pairwise to compute the cls_loss2. Is there any intuitive understanding of cls_loss2 and why it is effective?

**Comment by Hzzone** [comment]
1. The common few-shot object detection is what you think. I have not tried this idea. Current version is aligned with OV detection under the help of CLIP.
2. See "Hierarchical knowledge distillation" for motivation.
